src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -2482,43 +2482,43 @@ void MacroAssembler::debug64(char* msg, int64_t pc, int64_t regs[])
 #endif
     if (os::message_box(msg, "Execution stopped, print registers?")) {
       ttyLocker ttyl;
-      tty->print_cr(" pc = 0x%016lx", pc);
+      tty->print_cr(" pc = 0x" INT64_FORMAT, pc);
```

Comment:
This looks to me like we've changed from a hex representation to a decimal representation, which, given the 0x in front, doesn't seem like what is wanted.  I think the right thing to do is to either convert the whole "0x%16lx" to INTPTR_FORMAT or to use UINT64_FORMAT_X (which translates to PRIx64).

This applies to the register printing below as well.

Reply:
I totally trust battleblow + bsdkurt ;-) to find the right changes because my dtrace hack stopped working exactly here.
I remember a debug message like: "0x% 016lx is not an entry point" , or anything like that ...

Reply:
I used UINT64_FORMAT_X to avoid having to cast the arguments as well.